cube_stack_arm:
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50
  
  arm_position_controller:
    type: "position_controllers/JointGroupPositionController"
    joints: [arm_shoulder_pan_joint, arm_shoulder_lift_joint, arm_elbow_flex_joint, arm_wrist_flex_joint]
    #pid: {p: 100.0, i: 0.01, d: 10.0}
    publish_rate: 50
  
  arm_effort_controller:
    type: "effort_controllers/JointGroupEffortController"
    joints: [arm_shoulder_pan_joint, arm_shoulder_lift_joint, arm_elbow_flex_joint, arm_wrist_flex_joint, gripper_joint]
    publish_rate: 50
  
  arm_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints:
      - arm_shoulder_pan_joint
      - arm_shoulder_lift_joint
      - arm_elbow_flex_joint
      - arm_wrist_flex_joint
      - gripper_joint
    constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      arm_shoulder_pan_joint: {trajectory: 0.1, goal: 0.1}
      arm_shoulder_lift_joint: {trajectory: 0.1, goal: 0.1}
      arm_elbow_flex_joint: {trajectory: 0.1, goal: 0.1}
      arm_wrist_flex_joint: {trajectory: 0.1, goal: 0.1}
      gripper_joint: {trajectory: 0.1, goal: 0.1}
    stop_trajectory_duration: 0.5
    state_publish_rate:  25
    action_monitor_rate: 10